src/data/sum/interval.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.sum.order
+import order.locally_finite
+
+/-!
+# Finite intervals in a disjoint union
+
+This file provides the `locally_finite_order` instance for the disjoint sum of two orders.
+
+## TODO
+
+Do the same for the lexicographic sum of orders.
+-/
+
+open function sum
+
+namespace finset
+variables {α₁ α₂ β₁ β₂ γ₁ γ₂ : Type*}
+
+section sum_lift₂
+variables (f f₁ g₁ : α₁ → β₁ → finset γ₁) (g f₂ g₂ : α₂ → β₂ → finset γ₂)
+
+/-- Lifts maps `α₁ → β₁ → finset γ₁` and `α₂ → β₂ → finset γ₂` to a map
+`α₁ ⊕ α₂ → β₁ ⊕ β₂ → finset (γ₁ ⊕ γ₂)`. Could be generalized to alternative monads if we can make
+sure to keep computability and universe polymorphism. -/
+def sum_lift₂ : Π (a : α₁ ⊕ α₂) (b : β₁ ⊕ β₂), finset (γ₁ ⊕ γ₂)
+| (inl a) (inl b) := (f a b).map embedding.inl
+| (inl a) (inr b) := ∅
+| (inr a) (inl b) := ∅
+| (inr a) (inr b) := (g a b).map embedding.inr
+
+@[simp] lemma sum_lift₂_inl_inl (a : α₁) (b : β₁) :
+  sum_lift₂ f g (inl a) (inl b) = (f a b).map embedding.inl := rfl
+
+@[simp] lemma sum_lift₂_inl_inr (a : α₁) (b : β₂) : sum_lift₂ f g (inl a) (inr b) = ∅ := rfl
+@[simp] lemma sum_lift₂_inr_inl (a : α₂) (b : β₁) : sum_lift₂ f g (inr a) (inl b) = ∅ := rfl
+
+@[simp] lemma sum_lift₂_inr_inr (a : α₂) (b : β₂) :
+  sum_lift₂ f g (inr a) (inr b) = (g a b).map embedding.inr := rfl
```

Comment:
I think these will all be automatically generated if you just tag `sum_lift₂` with simp.